File: daq/slc/apps/hvcontrold/src/HVCommand.cc
#include "daq/slc/apps/hvcontrold/HVCommand.h"

#include "daq/slc/apps/hvcontrold/HVState.h"

using namespace Belle2;

const HVCommand HVCommand::CONFIGURE(101, "CONFIGURE", "CONFIGURE");
const HVCommand HVCommand::SAVE(102, "SAVE", "SAVE");
const HVCommand HVCommand::TURNON(103, "TURNON", "TURNON");
const HVCommand HVCommand::TURNOFF(104, "TURNOFF", "TURNOFF");
const HVCommand HVCommand::RAMPUP(105, "RAMPUP", "RAMPUP");
const HVCommand HVCommand::RAMPDOWN(106, "RAMPDOWN", "RAMPDOWN");
const HVCommand HVCommand::STANDBY(111, "STANDBY", "STANDBY");
const HVCommand HVCommand::STANDBY2(112, "STANDBY2", "STANDBY2");
const HVCommand HVCommand::STANDBY3(113, "STANDBY3", "STANDBY3");
const HVCommand HVCommand::PEAK(114, "PEAK", "PEAK");
const HVCommand HVCommand::RECOVER(121, "RECOVER", "RECOVER");

const HVCommand& HVCommand::operator=(const std::string& label) throw()
{
  if (label == CONFIGURE._label) *this = CONFIGURE;
  else if (label == SAVE._label) *this = SAVE;
  else if (label == TURNON._label) *this = TURNON;
  else if (label == TURNOFF._label) *this = TURNOFF;
  else if (label == RAMPUP._label) *this = RAMPUP;
  else if (label == RAMPDOWN._label) *this = RAMPDOWN;
  else if (label == STANDBY._label) *this = STANDBY;
  else if (label == STANDBY2._label) *this = STANDBY2;
  else if (label == STANDBY3._label) *this = STANDBY3;
  else if (label == PEAK._label) *this = PEAK;
  else if (label == RECOVER._label) *this = RECOVER;
  else if (label == Command::OK.getLabel()) *this = Command::OK;
  else if (label == Command::ERROR.getLabel()) *this = Command::ERROR;
  else *this = Enum::UNKNOWN;
  return *this;
}

const HVCommand& HVCommand::operator=(const char* label) throw()
{
  if (label != NULL)  *this = std::string(label);
  else *this = Enum::UNKNOWN;
  return *this;
}

File: daq/slc/extra/javalib/b2daq/src/b2daq/hvcontrol/core/HVCommand.java
package b2daq.hvcontrol.core;

import b2daq.nsm.NSMCommand;

public class HVCommand extends NSMCommand {

        final static public HVCommand UNKNOWN = new HVCommand(0, "UNKNOWN");
        final static public HVCommand CONFIGURE = new HVCommand(101, "CONFIGURE");
        final static public HVCommand TURNON = new HVCommand(102, "TURNON");
        final static public HVCommand TURNOFF = new HVCommand(103, "TURNOFF");
        final static public HVCommand STANDBY = new HVCommand(111, "STANDBY");
        final static public HVCommand SHOULDER = new HVCommand(112, "SHOULDER");
        final static public HVCommand PEAK = new HVCommand(113, "PEAK");
        final static public HVCommand RECOVER = new HVCommand(121, "RECOVER");
        final static public HVCommand HVAPPLY = new HVCommand(131, "HVAPPLY");

	public HVCommand() {
		copy(HVCommand.UNKNOWN);
	}

	private HVCommand(int id, String label) {
		super(id, label);
	}

	public HVCommand(HVCommand command) {
		copy(command);
	}

        @Override
	public boolean copy(String msg) {
		if (super.copy(msg)) { return true; }
		else if (msg.equals(CONFIGURE.getLabel())) { copy(CONFIGURE); return true; } 
		else if (msg.equals(TURNON.getLabel())) { copy(TURNON); return true; }
		else if (msg.equals(TURNOFF.getLabel())) { copy(TURNOFF); return true; }
		else if (msg.equals(STANDBY.getLabel())) { copy(STANDBY); return true; }
		else if (msg.equals(SHOULDER.getLabel())) { copy(SHOULDER); return true; }
		else if (msg.equals(PEAK.getLabel())) { copy(PEAK); return true; }
		else if (msg.equals(RECOVER.getLabel())) { copy(RECOVER); return true; }
		else if (msg.equals(HVAPPLY.getLabel())) { copy(HVAPPLY); return true; }
		return false;
	}

        @Override
	public boolean copy(int id) {
		if (super.copy(id)) { return true; }
		else if (id == CONFIGURE.getId()) { copy(CONFIGURE); return true; } 
		else if (id == TURNON.getId()) { copy(TURNON); return true; }
		else if (id == TURNOFF.getId()) { copy(TURNOFF); return true; }
		else if (id == STANDBY.getId()) { copy(STANDBY); return true; }
		else if (id == SHOULDER.getId()) { copy(SHOULDER); return true; }
		else if (id == PEAK.getId()) { copy(PEAK); return true; }
		else if (id == RECOVER.getId()) { copy(RECOVER); return true; }
		else if (id == HVAPPLY.getId()) { copy(HVAPPLY); return true; }
		return false;
	}

}
